client_test.go

```diff
@@ -257,3 +265,92 @@ func TestContextDeadline(t *testing.T) {
 		}
 	}
 }
+
+func TestRegionSubstitutionWithDefaultURL(t *testing.T) {
+	var host string
+	client := turbopuffer.NewClient(
+		option.WithAPIKey("tpuf_A1..."),
+		option.WithRegion("my-cool-region"),
+		option.WithHTTPClient(&http.Client{
+			Transport: &closureTransport{
+				fn: func(req *http.Request) (*http.Response, error) {
+					host = req.URL.Host
+					return &http.Response{
+						StatusCode: http.StatusOK,
+					}, nil
+				},
+			},
+		}),
+	)
+	client.Namespaces(context.Background(), turbopuffer.NamespacesParams{})
```

Comment:
It's a little odd to drop the error here instead of making some kind of assertion on it, but I see why you're doing it.